Feature

satana1volga opened the pull request:
Инит коммит был уже с написанным кодом, надеюсь что комментировать можно не только изменения, хотя там tsx файлы все как новые

satana1volga committed:
formated with prettier

satana1volga committed:
Added comments, started to use styled components instead of css modules

satana1volga committed:
fulfully created style components and deleted unnecessary css modules files

satana1volga committed:
added styled components

satana1volga committed:
new deps and prettier config

satana1volga committed:
перехожу на русский язык, переписал все на ts

satana1volga committed:
починил те проблемы что мне всякие анализаторы подсвечивали, бесило немного

satana1volga committed:
Добавил конфиг

satana1volga committed:
пофиксил еще проблемы которые подсветил линтер

satana1volga committed:
небольшой редизайн

sayonara20xx commented:
https://codesandbox.io/s/funny-bartik-wdbbve - sandbox feature

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+      // Костыль: у веб апи нельзя определить кол-во постов, получаем 500, выкидываю 400
+      .then((json) => {
+        setNewsIds(() => {
+          return json.slice(0, 100);
```
Новости не отсортированны по дате
![image](https://user-images.githubusercontent.com/44724024/179467510-5f96f32f-b709-499b-80f9-2510dadcdd76.png)

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/Comment/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+  return (
+    <Comm>
+      <NIcknameP>{commInfo.by}</NIcknameP>
+      <p onClick={commClicked}>{returnText()}</p>
```
Не рендерится `html` верстка в контенте коммента
![image](https://user-images.githubusercontent.com/44724024/179468051-03989b3f-3867-44d8-b1b4-cc1ceb48b857.png)

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+            </DivWrapper>
+          }
+        />
+        <Route path="/info" element={<NewsInfo data={selectedNew} />} />
```
Использовать `route Data` норм, но не всегда)
Например тут, зайдя на роут новости все ок, пока не обновишь страницу и не получишь вот это 
![image](https://user-images.githubusercontent.com/44724024/179468299-459ed33c-7d2f-4c4d-8136-40cd7d942f9d.png)
Поэтому `routeData` больше для каких то не `persist` вещей. Поищи тут другое решение как можно хранить и передавать информацию о выбранной новости

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+    });
+  }, [newsIds]);
+
+  const refreshButtonClick = (): void => {
```
`refresh button` сделан не совсем удобно.
При нажатии на него весь экран белым становится на секунду. Не очень `UX`.
Было бы классно если бы ты только ререндерил сами итемы

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/NewsInfo.tsx:
```diff
@@ -0,0 +1,139 @@
+          {"Rating: " + props.data.score + ", author: " + props.data.by}
+        </TitleText>
+        <TitleText>{"Publication date: " + formattedDate}</TitleText>
+        <TitleText>{"Descendants: " + props.data.descendants}</TitleText>
```
Не самое очевидное название для количества комментариев)
Мб по проще и дефолт `Comments` / `CommentsCount`?

Stilwuc commented on package.json:
```diff
@@ -6,11 +6,13 @@
     "@testing-library/react": "^13.3.0",
     "@testing-library/user-event": "^13.5.0",
-    "prettier": "2.7.1",
+    "eslint": "^8.19.0",
```
`eslint`, `prettier`, `testing utils` должны быть `devDependencies`, так как эти пакеты не будут использоваться в проде(чисто `dev ex`)

Stilwuc commented on public/index.html:
```diff
@@ -4,6 +4,7 @@
         <meta charset="utf-8">
         <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
         <meta name="theme-color" content="#000000">
+            <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">
```
Лучше `boostrap` тоже через `npm` подрубать, а то не камельфо когда что-то через `cdn'ку`, а что то через `npm`.
Да и вообщем то `npm` приоритетней для юза

Stilwuc commented on src/App.tsx:
```diff
@@ -1,12 +1,13 @@
+import React from "react";
 
-function App() {
+const App: () => JSX.Element = () => {
```
Лучше юзать type `alias` из `"react"` - `FC`.
Поддерживает передачу пропов и тд, и лучше привыкать юзать его, в `react 18` так точно)

Stilwuc commented on src/App.tsx:
```diff
@@ -1,12 +1,13 @@
-function App() {
+const App: () => JSX.Element = () => {
   return (
     <div className="App">
```
Не увидел стилей на этот класс. Можно выпилить вместе с дивом, лишних элементов в доме не будем создавать)

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/NewsInfo.tsx:
```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
```
Так как ты переписал модули на styled, то структура типа - `NewsSummary/NewsSummary.tsx` уже излишне, достаточно просто хранить `NewsSummary.tsx`.
Просто не стоит делать много вложенных папок, потом придется постоянно их раскрывать при поиске компонента)
Если что, группировать какую то логику имеет смысл, но тут пока нет

Stilwuc commented on src/HackerNews/HackerNews.tsx:
```diff
@@ -0,0 +1,15 @@
+const HackerNews: () => JSX.Element = () => {
+  return (
+    <BrowserRouter>
+      <div>
```
Тут точно лишний `div`. Без надобности доп вложенность `DOM` не делаем

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+
+  if (newsIds.length > 0) {
+    return (
+      <Routes>
```
По факту у тебя `MainPage` (текущий компонент) не является таковым, так как в нем создаются два других роута.
`Routes` предлагаю выкинуть повыше в `HackerNews`, и там объявить два роута `MainPage` и `InfoPage`.
\+ это пофиксит проблему условного рендеринга `Routes`, это не совсем хорошая практика рендерить список роутов по условию которое часто меняется, тем более тебе здесь это точно не нужно

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+
+import styled from "styled-components";
+
+export interface response {
```
`interface` и `type` в `PascalCase`
Ну и возможно название можно менее абстрактней сделать)
```suggestion
export interface NewsItem {
```

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+  by: string;
+  id: number;
+  descendants: number;
+  kids: Array<number>;
```
Лучше short синтаксис типизации массива юзать, легче читать)
```suggestion
  kids: number[];
```

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+  const url_new =
+    "https://hacker-news.firebaseio.com/v0/newstories.json?print=pretty";
+
+  const [news, setNews] = useState<Array<response>>(() => {
```
Необязательно тут юзать `getter` для начального значения, проще будет обычной иницилизацией
```suggestion
  const [news, setNews] = useState<Array<response>>([]);
```

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+  const [selectedNew, setSelectedNew] = useState<response>(() => {
+    return {
+      by: "",
+      id: -1,
```
Не стоит делать empty объект под интерфейс, boilerplate код плодится. 
Лучше сделай что стейт может быть как `response` так и `null`

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+        });
+      });
+
+    setInterval(() => {
```
Сейчас у тебя этот `setInterval` не чистится, и будет работать даже когда уйдем со страницы.
Обычно это ведет к утечке памяти, поэтому надо за собой все это подчищать перед уходом

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+      .then((response) => response.json())
+      // Костыль: у веб апи нельзя определить кол-во постов, получаем 500, выкидываю 400
+      .then((json) => {
+        setNewsIds(() => {
```
Тут `setter` тоже проще обычный юзать, так как все равно аргумент текущего стейта не юзаешь
```suggestion
        setNewsIds(json.slice(0, 100));
```

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+import React, { useEffect, useState } from "react";
```
Еще, не заюзал `state management`, лучше подумать что в него выкинуть и заюзать. Советую `createContext`, попроще)
![image](https://user-images.githubusercontent.com/44724024/179488622-467d5a3e-d33c-4c11-9964-bfce56441955.png)

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+`;
+
+const MainPage: () => JSX.Element | null = () => {
+  const url =
```
Названия не делай слишком абстрактные. Пример смотрю я такой `fetch(url)` и сейчас по названию переменной я 0 идей куда запрос пойдет. В данном случае например можно назвать `fetchTopStoriesUrl`
![image](https://user-images.githubusercontent.com/44724024/179489898-bb86f449-ed08-43ca-8e88-db75edfc85f8.png)

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+      });
+
+    setInterval(() => {
+      if (window.location.pathname === "/") {
```
От этой прикольной проверки уйдем, если разделить страницы на разные компоненты

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+      fetch(detailsUrl)
+        .then((response) => response.json())
+        .then((json) => {
+          setNews((prevState) => {
```
Интересное решение, у тебя 100 раз вызывается `setState`. По идее `batching` сработает и кучу ререндеров не будет лишних, но все же мне кажется лучше один раз засетить 100 элементов чем 100 раз один)

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+
+    fetch(url)
+      .then((response) => response.json())
+      // Костыль: у веб апи нельзя определить кол-во постов, получаем 500, выкидываю 400
```
Не надо дублировать одну и ту же функцию `fetch` два раза. Вынеси получение `id` новостей в отдельную функцию и вызывай её тут и в `setInterval`

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+  const [news, setNews] = useState<Array<response>>(() => {
+    return [];
+  });
+  const [newsIds, setNewsIds] = useState(() => {
```
Типизации нету. Сейчас это `never[]`
```suggestion
  const [newsIds, setNewsIds] = useState<number[]>(() => {
```

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+      </Routes>
+    );
+  } else {
+    return null;
```
Чтобы не было больших вложенных блоков if, лучше юзать ранний выход из функции, в твое примере это было бы так
```
if (!newsIds.length) return null;

return (
      <Routes>
....
```

Stilwuc commented on src/HackerNews/MainPage/MainPage.tsx:
```diff
@@ -0,0 +1,192 @@
+                    data={elem}
+                    number={index}
+                    select={setSelectedNewCallback}
+                  ></NewsSummary>
```
Юзай самозакрывающие теги если children не нужен
```suggestion
                  />
```

Stilwuc commented on tsconfig.json:
```diff
@@ -0,0 +1,105 @@
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+
+    /* Projects */
+    // "incremental": true,                              /* Save .tsbuildinfo files to allow for incremental compilation of projects. */
```
в json комментов нету, возможно tsconfig пофиг, но лучше выпилить, тем более что комменты не особо полезные, это всегда можно в доке посмотреть)

Stilwuc commented on src/HackerNews/MainPage/NewsSummary/NewsSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+  select: (event: React.MouseEvent<HTMLAnchorElement>, obj: response) => void;
+}
+
+const NewsSummary: (arg0: props) => JSX.Element = (props) => {
```
Такая типизация компонента не очень, через FC здесь было бы так
```suggestion
const NewsSummary: FC<props> (props) => {
```
P. S. Не забудь кейс интерфейса поправить

Stilwuc commented on src/HackerNews/MainPage/NewsSummary/NewsSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+  margin: 0 auto 10px auto;
+`;
+
+export interface props {
```
Так как ты этот интерфейс еще экспортишь, надо точно норм название для него. А то прикинь у тебя будет 10 таких интерфейсов абстрактных из разных компонентов экспортится, не удобно.
Обычно название интерфейса пропов это "Название компонента + Props", поэтому тут получится `NewsSummaryProps`

Stilwuc commented on src/HackerNews/MainPage/NewsSummary/NewsSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+`;
+
+export interface props {
+  data: {
```
Ты этот интерфейс уже объявлял, лучше переиспользовать его

Stilwuc commented on src/HackerNews/MainPage/NewsSummary/NewsSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+  return (
+    <DivContainer>
+      <p>{props.number + 1 + ". " + props.data.title}</p>
+      <p>{"Rating: " + props.data.score + ", author: " + props.data.by}</p>
```
Конкатинацию строк лучше через интерполяцию делать. Лучше читается
```suggestion
      <p>{`Rating: ${props.data.score}, author: ${props.data.by}`}</p>
```

Stilwuc commented on src/HackerNews/MainPage/NewsSummary/NewsSummary.tsx:
```diff
@@ -0,0 +1,46 @@
+}
+
+const NewsSummary: (arg0: props) => JSX.Element = (props) => {
+  let formattedDate: Date = new Date(props.data.time * 1000);
```
Почему `let`, если значение не меняешь, юзай `const`

Stilwuc replied:
Ты кстати не форматнул дату, типо в текущем варианте он просто Date.toString делает, и получается не совсем красиво. Лучше бы в более читаемом формате выводить)
![image](https://user-images.githubusercontent.com/44724024/179495365-2b816f4d-247f-4031-b0f1-79f1062ea121.png)

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/Comment/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+  margin-top: 50px;
+`;
+
+const NIcknameP = styled.p`
```
Странный кейс названия.
\+ необязательно в конце постфиксом добавлять тип элемента(`P`), иногда достаточно просто `Nickname`

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/Comment/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+  font-size: 20px;
+`;
+
+export interface CommProps {
```
Сокращения юзай обдуманно.
сокращения `comm` для comments неочевидно, а если не очевидно, то лучше и не сокращать.
Читаемость > длины названия

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/Comment/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+  }, [props.id]);
+
+  const returnComms: () => JSX.Element | null = () => {
+    if (isClicked && commInfo.kids && props.id && commInfo.kids.length > 0)
```
Вот такой баг славил на раскрытие дерева комментов
![image](https://user-images.githubusercontent.com/44724024/179496219-cd99054f-3923-49bd-99f1-e50ea9cfed3d.png)
Id = 32134691

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/Comment/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+    if (!commInfo.kids) {
+      return commInfo.text;
+    } else {
+      return commInfo.text + ". Click on text to watch reply(es).";
```
Чутка придирка, но неочевидно что на нажатие откроется тред комментов, так как слипается с текстом коммента
![image](https://user-images.githubusercontent.com/44724024/179496487-fbe3cf5e-2a73-4902-8806-68cf62da920f.png)

Stilwuc commented on src/HackerNews/MainPage/NewsInfo/Comment/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+    };
+  });
+
+  const [isClicked, setIsClicked] = useState(() => false);
```
При обновлении комментов по времени(раз в минуту), сбрасывается все состояние открытых комментов, и скролимся в топ. Не удобно(

Stilwuc reviewed (changes requested):
Вообщем неплохо.
Пару важных моментов поправь и вообще хорошо будет :)

crutch12 commented:
С дизайном всё очень плохо, выкидывай все свои заготовки
и сажай на react-bootstrap или mui

satana1volga committed:
Добавил конфиг для форматтера кода

satana1volga committed:
Удалил комментарии

satana1volga committed:
Перекинул в dev-dependencies все то, что не должно идти в прод

satana1volga committed:
Удалил bootstrap из htmlки

satana1volga committed:
Начал поправлять ошибки:
вынес повыше роутер и передал ему в аргументы нормальную компоненту,
подправил ts типы, начал использовать контекст,
пофиксил ошибку рендера при нулевом количестве комментов,
вынес все интерфейсы в отдельный файл чтобы разбить цикличный импорт, интерполировал текст в html каркас для комментариев,
переписал обновление новостей так что теперь отрисовка идет после получения инфы о всех ста новостях,
удалил лишние папки и лишнюю вложенность элементов

satana1volga committed:
Начал сохранять выбранную новость для отображения в NewsInfo в localStorage и сам коллбэк для этого положил в контекст вместе с стейтом для хранения выбранной новости,
Попытался сделать обновляющиеся комментарии с сохранением структуры раскрытия дочерних но вышло обновление лишь корневых,
Добавил удаление интервалов через их ID,
убрал страшную и резкую очистку всего окна при обновлении новостей

satana1volga committed:
Добавил файлы .idea в gitignore

satana1volga committed:
Поменял дизайн, выпилив все styled components и заменив их bootstrap,
Исправил скачущую надпись с открытием дочерних комментариев, которая появлялась и исчезала когда их не было,
Исправил значение ключа внутри map на id новостей, с сервера иногда приходили одни и те же заголовки,
Новости теперь сортируютс точно по времени

satana1volga committed:
Сменил тип на React.FC

sayonara20xx commented:
Вроде как все подправил, запилил новый дизайн, новости в виде бутстраповского грида, вся инфа в виде его же карточек, надеюсь так будет получше.

Обновление комментариев с сохранением вида развернутых пытался реализовать двумя способами: 
передавать им в пропсы их детей (не представил как обновлять чисто какой-то стейт дочерний и при этом все не перерендерить);
с помощью доп инфы в виде коллекции данных в контексте (объект, который обновляется каждый раз при изменении стейта isClicked в каждом коменте, соответственно каждый комментарий должен знать свое место в дереве, эту инфу можно внутри map сформировать и передавать так же через props, но я не смог нормально догнать как этот объект модифицировать, по сути надо было превратить массив в данных в последовательность ключей для обращения к этому объекту, но это как-то все громоздко)

Еще, интервалы я чистил только при использовании NavLink, не знаю, нужно ли еще как-то их отрабатывать. И насчет использования localStorage для хранения выбранной новости тоже не уверен, но с ним все работает.

sayonara20xx commented:
https://codesandbox.io/s/pensive-surf-t8l6iw - codesandbox

Stilwuc commented on src/index.js:
```diff
@@ -3,7 +3,7 @@ import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
 
+import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
```
При импорте либ обычно `node_modules` не надо, должно заресолвится само
```suggestion
import 'bootstrap/dist/css/bootstrap.min.css';
```

Stilwuc commented on src/HackerNews/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+import { CommentInfo, CommentProps } from './Interfaces';
+import { Card } from 'react-bootstrap';
+
+const HighlightedP = styled.p`
```
Аккуратно с `<p>`. Например сейчас у тебя есть вложенные параграфы в другие параграфы, а это html не корретно
![image](https://user-images.githubusercontent.com/44724024/180993193-2d7f8d0c-dd56-4682-82a7-110308bcabf4.png)

Stilwuc commented on src/HackerNews/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+    return null;
+  });
+
+  const [commentKids, setCommentKids] = useState<number[] | null>(() => {
```
Не везде это поправил
![image](https://user-images.githubusercontent.com/44724024/180994003-20640268-ee9c-41a2-bd93-6a2b69945b31.png)

Stilwuc commented on src/HackerNews/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+      <Card.Body>
+        <Card.Title>{commentInfo?.by}</Card.Title>
+        <Card.Text onClick={commClicked}>{returnText()}</Card.Text>
+        {renderComments()}
```
Было бы неплохо если бы ты deleted комменты обрабатывал
![image](https://user-images.githubusercontent.com/44724024/180994228-da87c322-4a96-4791-9abe-aba54b8d7fd9.png)

Stilwuc commented on src/HackerNews/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+    fetch(url)
+      .then((response) => response.json())
+      .then((json) =>
+        setCommentInfo(() => {
```
можешь просто ложить весь `response` в `setter`, даже если будет больше полей чем ожидает state все ок, до тех пор пока будут те обязательные поля

Stilwuc commented on src/HackerNews/Comment.tsx:
```diff
@@ -0,0 +1,99 @@
+  }, [props.id]);
+
+  useEffect(() => {
+    let url = `https://hacker-news.firebaseio.com/v0/item/${props.id}.json?print=pretty`;
```
Ты делаешь два одинаковы запроса(#24) чтобы результат пополам поделить и положить в разные стейты, не оч.
Ты в одном запросе можешь сделать два сетера, да и вообще два стейта тут не зачем

Stilwuc commented on src/HackerNews/SelectedNewStore.ts:
```diff
@@ -0,0 +1,10 @@
+
+interface NewStore {
+  selectedNew: React.ComponentState;
+  setSelectedNewCallback: (event: React.MouseEvent<HTMLAnchorElement>, obj: NewsDataResponse) => void;
```
Обычно наименование коллбэков с префиксом `on` идет, ты наверно видел такое в тех же `<button>` и тд.
Поэтому тут можно просто что то типо
```suggestion
  onSelectNews: (event: React.MouseEvent<HTMLAnchorElement>, obj: NewsDataResponse) => void;
```

Stilwuc commented on src/HackerNews/MainPage.tsx:
```diff
@@ -0,0 +1,107 @@
+
+const MainPage: React.FC = () => {
+  const newStoriesUrl = 'https://hacker-news.firebaseio.com/v0/newstories.json?print=pretty';
+  const IntervalID: React.MutableRefObject<NodeJS.Timer | undefined> = useRef();
```
интересно что это в PascalCase хотя это обычная переменная)

Stilwuc commented on src/HackerNews/MainPage.tsx:
```diff
@@ -0,0 +1,107 @@
+
+  useEffect(() => {
+    fetchNewsIdsByUrl(newStoriesUrl).then((json) => {
+      setNewsIds(() => {
```
Тоже можно избавиться от второго стейта, просто здесь сразу выполни функцию фетча по новости по id
А то контролировать два стейта отвечающее за +- одно и то же сложней чем один)

Stilwuc commented on src/HackerNews/MainPage.tsx:
```diff
@@ -0,0 +1,107 @@
+        .then((response) => response.json())
+        .then((json) => {
+          tempArray = [...tempArray, json];
+          if (tempArray.length === 100) {
```
Ты можешь заюзать `Promise.all` с масивом промисей получившихся через `map`, подождать когда они заресолвятся и после этого делать sort и сет.
А то хардкод длина массива = 100 значит элемент последний не кайф

Stilwuc commented on src/HackerNews/MainPage.tsx:
```diff
@@ -0,0 +1,107 @@
+
+  return (
+    <Container>
+      <Row className={'justify-content-md-center mt-3'}>
```
```suggestion
      <Row className="justify-content-md-center mt-3">
```

Stilwuc commented on src/HackerNews/NewsInfo.tsx:
```diff
@@ -0,0 +1,108 @@
+  let selectedNew = store?.selectedNew;
+
+  if (!selectedNew) {
+    selectedNew = localStorage.getItem('selectedNewObject');
```
мм, интересное решение, но я все равно могу сломать его)
1. Открыть в одной вкладке новость 1
2. Открыть в другой вкладке новость 2
3. Обновить вкладку с новостью 1
4. После обновления ты получишь новость 2
Так что надо по другому идти, небольшая подсказка - это делается через react-router

Stilwuc commented on src/HackerNews/NewsInfo.tsx:
```diff
@@ -0,0 +1,108 @@
+    }, 60000);
+  }, []);
+
+  const refreshOnCLick = (): void => {
```
```suggestion
  const refreshOnClick = (): void => {
```

crutch12 commented:
Я вспомнил, что тебе нужно было сделать ещё опциональное задание
> Из опциональных, сделай:
> > Конфигурация сборки проекта в Docker image

satana1volga committed:
Добавил и проверил докерфайл, после билда с ним создается образ, который после запуска с расшарингом 3000 порта дает зайти на сайт

satana1volga committed:
Поправил названия переменных и некоторые типы ts,
сократил количество состояний для отображения комментариев и новостей (загрузка новостей раньше шла через промежуточный стейт, который хранил ID новостей и после обновления которого обновлялся другой стейт, хранящий уже сами новости, в комментариях инфа из одного запроса была в разных стейтах, теперь в одном, весь код переписан с упором на это),
переписал получение новостей на промисы (вместо захардкоженной сотки ждал, когда соберется весь массив промисов (fetch инфы о каждой новости), затем сбрасывал всю инфу в state)
обновил соответствующие интерфейсы в связи с изменениями сверху
чтобы после перезагрузки в каждом копии приложении сохранялась выбранная новость использую параметр в url, который используется если !selectedNew

satana1volga committed:
Добавил никому не нужный файл

satana1volga committed:
Удалил никому не нужную папку

Stilwuc commented on src/HackerNews/MainPage.tsx:
```diff
@@ -0,0 +1,98 @@
+  };
+
+  const clearNews = () => {
+    setNews(() => {
```
Уже писал, еще раз напишу
![image](https://user-images.githubusercontent.com/44724024/183043902-ebbc08ab-6b32-47ed-92c3-87e1af961e50.png)

Stilwuc commented on src/HackerNews/NewsInfo.tsx:
```diff
@@ -0,0 +1,117 @@
+    });
+
+    IntervalID.current = setInterval(() => {
+      refreshOnClick();
```
При авторевреше начинаются проблемы
![image](https://user-images.githubusercontent.com/44724024/183080060-ef85ffd5-77d8-4b3f-891d-c3bf4d0da308.png)

sayonara20xx commented:
Все подправил

Stilwuc commented on src/HackerNews/Comment.tsx:
```diff
@@ -0,0 +1,89 @@
+        <Card.Body>
+          <Card.Title>{commentInfo?.by}</Card.Title>
+          <span onClick={commClicked}>{returnText()}</span>
+          {renderKidsComments()}
```
На f5 комменты сразу не рендерятся.
Только если нажму refreshComments
![image](https://user-images.githubusercontent.com/44724024/183078948-6df4bcfb-aff7-47fa-9ef9-b0edebacc8d6.png)
![image](https://user-images.githubusercontent.com/44724024/183079010-2fa1c73a-4daa-4e62-8a57-2a850e51742a.png)

Stilwuc reviewed (changes requested):
Точно ластовые)
Докер для дева подет

satana1volga committed:
Исправил ошибку при автообновлении комментариев и исправил их поведение при ручном обновлении страницы, добавил обработку "мертвых" комментариев (помимо isDeleted у API есть ключ isDead)

satana1volga committed:
Исправил еще одну ошибку

sayonara20xx commented:
Поправил, тупые ошибки были
https://codesandbox.io/s/infallible-kapitsa-951ubv - codesandbox

Stilwuc commented:
Вот теперь шикарно все 👍 
@sayonara20xx Можно мержить
